scrapyelasticsearch/scrapyelasticsearch.py

```diff
@@ -158,3 +163,25 @@ def close_spider(self, spider):
         if len(self.items_buffer):
             self.send_items()
 
+    def create_index_with_mapping(self, ext):
+        index_name = self.create_index_name(ext)
+        mapping_file = ext.settings['ELASTICSEARCH_INDEX_MAPPING']
+
+        if ext.es.indices.exists(index_name):
+            logging.debug("index already exists, mapping will not be updated")
+        else:
+            logging.debug("will create index with mapping definition")
+            file = open(mapping_file, "r")
+            mapping_data = json.loads(file.read())
+            ext.es.indices.create(
+                index=index_name, ignore=400, body=mapping_data)
+
+    def create_index_name(ext):
```

Comment:
Not mandatory  but I would extrapolate from `index_item` index name computation and rename this method like `_get_index_name`.